cmake/configure/configure_boost.cmake

```diff
@@ -87,6 +87,12 @@ ENDMACRO()
 MACRO(FEATURE_BOOST_FIND_EXTERNAL var)
   FIND_PACKAGE(BOOST)
 
+  IF(NOT "${BOOST_DIR}" STREQUAL "" AND NOT BOOST_FOUND)
+    MESSAGE(ERROR
+            "Could not find Boost in the provided prefix"
+           )
+  ENDIF()
```

Comment:
This error message fires exactly if no external boost is found and `BOOST_DIR` is non-empty.

This is very misleading due to a couple of reasons. First, it does not imply that the external boost that is found is actually `BOOST_DIR`.

Secondly, this contradicts the current notion of `<FEATURE>_DIR` being a hint. So if we change that for boost, we have to change it for every feature (which will be a mess).

Third, if you do not want to pick up bundled libraries, simply set `DEAL_II_ALLOW_BUNDLED` to false.